tools/autograd/templates/variable_factories.h

```diff
@@ -22,80 +22,170 @@ using at::DimnameList;
 namespace torch {
 
 namespace detail {
-  enum class InitListTensorType { Scalar, InitList };
-
-  // We use `InitListTensor` to support converting an arbitrarily nested braced-init-list
-  // (e.g. {{1, 2}, {3, 4}}) into the equivalent Tensor, taking advantage of the fact that
-  // the constructor will automatically be called recursively until it reaches all innermost
-  // scalar values.
-  //
-  // At any time, a `InitListTensor` object represents either of the following:
-  // 1. A scalar with value `scalar()` and type `scalar_type()`.
-  // 2. A Tensor represented in `std::initializer_list<InitListTensor>` form, with value
-  //    `init_list()`, Tensor scalar type `scalar_type()`, and Tensor sizes `sizes()`.
-  struct InitListTensor {
-#define TENSOR(T, S)                   \
-    InitListTensor(T scalar) :         \
-        scalar_(scalar), init_list_(), \
-        sizes_(),                      \
-        scalar_type_(at::k##S),        \
-        type_(InitListTensorType::Scalar) {}
+
+#define NESTED_INIT_LIST_MAX_DEPTH 10
```

Comment:
This is better as `constexpr int64_t kNestedInitListMaxDepth = 10`

Reply:
Thanks! I refactored the PR and got rid of the need for this const variable.